data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -168,18 +168,21 @@ public void start(Buffer<Record<Event>> buffer) {
     @Override
     public void stop() {
         shutdownInProgress.set(true);
-        LOG.info("Shutting down Consumers...");
-        sourceConfig.getTopics().forEach(topic -> {
-            stopConsumer(topicExecutorService.get(topic), topicConsumer.get(topic));
-        });
-        LOG.info("Consumer shutdown successfully...");
+        final long shutdownWaitTime = calculateLongestThreadWaitingTime();
```

Comment:
What's the upper limit on this?

Reply:
Looking at calculateLongestThreadWaitingTime() implementation it looks like it can be atmost 5 seconds (default value) unless user has configured to be higher number.